Small improvements to debug prints

ujfalusi opened the pull request:
Hi,

Few minor improvements for debugging
- print out the internal fw_state value along with DSP dump and string which can help diagnose issue
- It has been observed that an IPC timeout is very well expected to happen time to time but it 'muted' subsequent dumps in case we were able to send IPC message without problems later.
- Use a macro to change the fw_state and add dev_dbg print for the state change
- drop the 'error' prefix for HDA dump prints.

dbaluta commented on sound/soc/sof/sof-priv.h:
```diff
@@ -586,6 +586,19 @@ static inline void sof_oops(struct snd_sof_dev *sdev, void *oops)
+		return;
+
+	dev_dbg(sdev->dev, "New fw_state: %d\n", new_state);
+	sdev->fw_state = new_state;
```
Maybe worth printing the old state too.

ujfalusi replied:
OK, I'll print that as well.

kv2019i reviewed (approved):
Looks good! Adding the old->new state to the fw-state printout is a good addition. But other than that, no concern. The re-enabling of dumps after a successful IPC TX, is a nice addition.

ujfalusi commented:
@dbaluta, @kv2019i, I have the change locally and going to push it on Monday.

plbossart commented on sound/soc/sof/ipc.c:
```diff
@@ -267,6 +267,12 @@ static int tx_wait_done(struct snd_sof_ipc *ipc, struct snd_sof_ipc_msg *msg,
+		if (sdev->ipc_dump_printed) {
+			sdev->dbg_dump_printed = false;
+			sdev->ipc_dump_printed = false;
+		}
```
I wasn't able to follow the commit message and comment. Usually an iPC timeout is rather fatal, audio will be broken until the next suspend-resume, it's not just a harmless 'hiccup'.

ujfalusi replied:
@plbossart, this was my understanding as well. I had other patches (introducing the FW_CRASHED state) which would declare the firmware crashed if IPC timeout happens, which I thought was a correct way to handle this.
Then I saw sever logs where we had an IPC timeout reported (should be fatal) but consequent IPC messages were transmitted without issue and everything was just fine. It was a hiccup on the firmware side (scheduling spike).
My initial idea was to _block_ all further IPC tx after a timeout, but after hunting for more logs I have seen others where the timeout was followed by a recovery.

The reason for the re-enabling the dumps is exactly for this case, we got a timeout, we dump, then we will not show any dump for the consequent timeouts (unless all dump is enabled). If any consequent IPC tx manages to succeed then we re-enable the dumps to catch the next timeout.

No functionality change, just to give us more information, possibly hinting to a rootcause.

plbossart replied:
I guess it depends where the IPC timeout happens. if it's for some random control that's probably ok. If it happens while restoring the state in the resume callback then it's as good as dead.

ujfalusi replied:
> I guess it depends where the IPC timeout happens. if it's for some random control that's probably ok. If it happens while restoring the state in the resume callback then it's as good as dead.

Yes, that's correct. But it could be that in case of random control we are also dead. Based on what I saw in logs, it is not consistent.
I would expect the I'm dead interrupt from firmware in case it things go that bad, but for the case when we got timeout, but the firmware is still OK we want to capture the next time it happens again.
Let me try to clarify the commit message.

plbossart reviewed (commented):
looks mostly good except for one set of comments below where you lost me @ujfalusi

ujfalusi commented:
Changes since v1:
- print both the old and new fw_state when it is changing
- new commit message to explain the reasons why to re-enable the dumps after a successful IPC tx
- new patch to force _all_ DSP crash dumps to be visible [1]
- new patch to clean up the prints in snd_sof_get_status()

[1] if for some reason the DSP panic interrupt comes later than we had IPC timout the DSP dump would have been disabled and we might miss crucial information to rootcause the problem.

ranj063 commented:
SOFCI TEST

ujfalusi commented:
Changes since v2:
- rebased on topic/sof-dev to trigger a new SOF CI run

ujfalusi commented:
Is there any comment that I have not addressed for this PR?
This and the #3170 would be a nice improvement for bug reports.

Changes since v3:
- rebased on topic/sof-dev to trigger a new SOF CI run

plbossart reviewed (approved):
thanks @ujfalusi
